app/views/admin/applications/show.html.erb

```diff
@@ -0,0 +1,20 @@
+<%= render partial: "layouts/application_render" %>
+
+<h4>Pets Applied For</h4>
+<% @application.pets.each do |pet| %>
+  <div id="pet-<%= pet.id %>">
+    <%= link_to "#{pet.name}", "/pets/#{pet.id}" %><br>
+    <% if @application.find_if_adoptable(pet.id) == false && @application.find_pet_application_status(pet.id) == "Pending" %>
+      <p>This pet has been approved for adoption</p><br>
+      <%= button_to "Reject Application", "/admin/applications/#{@application.id}?pet_id=#{pet.id}&button=reject", method: :patch %>
+    <% elsif @application.find_pet_application_status(pet.id) == "Approved" %>
+      <p>Application Approved</p>
+    <% elsif @application.find_pet_application_status(pet.id) == "Rejected" %>
+      <p>Application Rejected</p>
+    <% elsif @application.find_pet_application_status(pet.id) == "Pending" %>
```

Comment:
This view is going to be pretty complex, so none of these refactors make a huge world of a different but here are some ideas:

- First IF branch: `@application.find_pet_application_status(pet.id) == "Pending"`
- - - inside this if branch, check for `@application.find_if_adoptable(pet.id) == false` to indicate this pet isn't available
- - - otherwise add buttons
- ELSE
- - - render the appropriate text based on Approved/Rejected/Pending. You could even have this logic in a model method and call this method here because all you're rendering is text. Something like
- - - `<%= @application.find_pet_app_status_text_label(pet.id) %>`  which will return Approved or Rejected